lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -282,41 +304,56 @@ def configure(conf)
                 fetch_gce_metadata('instance/attributes/gae_backend_name')
             @gae_backend_version =
                 fetch_gce_metadata('instance/attributes/gae_backend_version')
-            @service_name = APPENGINE_SERVICE
-            common_labels["#{APPENGINE_SERVICE}/module_id"] = @gae_backend_name
-            common_labels["#{APPENGINE_SERVICE}/version_id"] =
-              @gae_backend_version
+            @resource.type = APPENGINE_RESOURCE_TYPE
+            @resource.labels['module_id'] = @gae_backend_name
+            @resource.labels['version_id'] = @gae_backend_version
           elsif attributes.include?('kube-env')
             # Kubernetes/Container Engine
-            @service_name = CONTAINER_SERVICE
-            common_labels["#{CONTAINER_SERVICE}/instance_id"] = @vm_id
+            @resource.type = CONTAINER_RESOURCE_TYPE
             @raw_kube_env = fetch_gce_metadata('instance/attributes/kube-env')
             @kube_env = YAML.load(@raw_kube_env)
-            common_labels["#{CONTAINER_SERVICE}/cluster_name"] =
+            @resource.labels['cluster_name'] =
               cluster_name_from_kube_env(@kube_env)
             detect_cloudfunctions(attributes)
           end
         end
-        common_labels["#{COMPUTE_SERVICE}/resource_type"] = 'instance'
-        common_labels["#{COMPUTE_SERVICE}/resource_id"] = @vm_id
+        # Some services have the GCE instance_id and zone as MonitoredResource
+        # labels; for other services we send them as entry labels.
+        if @resource.type == COMPUTE_RESOURCE_TYPE ||
+           @resource.type == CONTAINER_RESOURCE_TYPE
+          @resource.labels['instance_id'] = @vm_id
+          @resource.labels['zone'] = @zone
+        else
+          common_labels["#{COMPUTE_SERVICE}/resource_id"] = @vm_id
```

Comment:
As I understand it, outside of the resource labels, all common labels are one big shared pool. This prefixing actually provides a namespace for these metrics, so you could search the logs for, e.g., Cloud Functions version independently from some other, user-defined notion of version.
Are we going to recommend that users use the same, e.g., "version" label for everything? That would also clash with them defining their own label from some payload field, because it'll override the one we compute automatically. Not saying this is the wrong approach, but we do need to tread carefully.
Thoughts?